### ISSUE TITLE
UUID Plug-in

### ISSUE THREAD

**Issue by scottfurry** [sqlitebrowser/sqlitebrowser#1951]
Describe the new feature
--------------------------
Determine requirements and use cases for incorporating UUID plug-in to DB4S distribution.

Additional info
---------------
#### Is your feature request related to an issue? Please include the issue number.
From bug #1950. User was attempting to create a table involving UUID's. Research `tripped` over URL to github repo with detailed UUID capabilities for SQLite.

Link: https://github.com/benwebber/sqlite3-uuid

#### Does this feature exist in another product or project? Please provide a link.
Doubtful. Brief search on net did not show UUID ability as part of SQLite. New territory for us.

@justinclift  - FYI

**Comment by justinclift** [comment]
Yep, we should have a go at incorporating this. :smile:

**Comment by scottfurry** [comment]
Top of my head: this would require pulling in external code (not hard), tweaking build files (minor sweat factor), and setting up the UI to accept user input for UUID column type (insert heart palpitations here).

I don't know about 3.12... No pressure...much...

**Comment by justinclift** [comment]
We haven't set a tentative ETA for 3.12 yet.  Which we should probably get around to doing. :wink:

**Comment by mgrojo** [comment]
A good case for exploring git submodules as it should be an optional extension and thus, won't prevent compilation of DB4S itself?

By the way:

> This extension only supports Linux and Mac OS X at the moment.

**Comment by karim** [comment]
> This extension only supports Linux and Mac OS X at the moment.

True. I tried to build it for Windows but it didn't work. It includes/depends on `libuuid` which isn't available for Windows.

**Comment by justinclift** [comment]
> A good case for exploring git submodules

Hmmm.  If you want to try them out, please do it in a different repo.

Just because with git submodules, if we enable them for DB4S and later on decide it was a bad idea... then too bad, it's too late.  "There's no going back" with them kind of thing. :smiling_imp:

**Comment by scottfurry** [comment]
> By the way:
> 
> > This extension only supports Linux and Mac OS X at the moment.

Missed that caveat at the bottom of the page. Looks like libuuid is apart of software for e2fsprogs (ext2 stuff maintained by Theodore T'so). On top of that, the code is in C. Again, use case and requirements. Doesn't mean we can't "borrow" the ideas and implement ourselves. Just means more work.
